src/graph.rs

```diff
@@ -289,8 +290,55 @@ where
   }
 }
 
-fn is_false(v: &bool) -> bool {
-  !v
+#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
+pub enum DependencyKind {
+  /// The dependency was provided as an explicit import into the graph, either a
+  /// root module or via a synthetic import
+  Explicit,
+  /// The dependency was identified via an `import` or `import type` statement
+  Import,
+  /// The dependency was identified via an `export` or `export type` statement
+  Export,
+  /// The dependency was identified via a dynamic `import()` statement
+  DynamicImport,
+  /// The dependency was identified via a `require` statement
+  Require,
+  /// The dependency was identified via a pragma within the file
+  Pragma,
+}
+
+impl Default for DependencyKind {
+  fn default() -> Self {
+    Self::Import
+  }
+}
```

Comment:
Is it required to implement `Default` trait for this enum? It feels to me like a potential footgun that might lead to discrepancies by too liberal use of `Default::default()` somewhere in the call chain.